### DISCUSSION TITLE
What is a Magic Cookie ?

### DISCUSSION THREAD

**Discussion by oblesh6** [COVESA/vsomeip#720] [Q&A] [answered]
Could anyone explain what's a Magic Cookie is?
Why/What is the use of magic cookie in tcp stream? 
Its cons and pros. 

Any source to read about it

Thanks

**Comment by duartenfonseca** [comment, marked as answer]
hi @oblesh6 , here are two sources of info regarding this: 
https://github.com/COVESA/vsomeip/issues/291#issuecomment-987796005
https://some-ip.com/papers/cache/AUTOSAR_TR_SomeIpExample_4.2.1.pdf
